Instagram/View Controllers/HomeFeedViewController.m

```diff
@@ -0,0 +1,207 @@
+                [self.posts addObjectsFromArray:posts];
+            }
+            else {
+                self.posts = (NSMutableArray *) posts;
+            }
+            [self.tableView reloadData];
+        }
+        else {
+            NSLog(@"😫😫😫 Error getting home timeline: %@", error.localizedDescription);
+        }
+        [self.refreshControl endRefreshing];
+        [MBProgressHUD hideHUDForView:self.navigationController.view animated:YES];
+    }];
+}
+- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
+    if([segue.identifier isEqualToString:detailSegue]) {
+        PostTableViewCell *tappedCell = sender;
+        DetailsViewController *detailsViewController =  [segue destinationViewController];
+        detailsViewController.post = tappedCell.post;
+        detailsViewController.user = self.user;
+    }
+    else if([segue.identifier isEqualToString:profileSegue]) {
+        ProfileViewController *profileViewController =  [segue destinationViewController];
+        profileViewController.user = sender;
+    }
+}
+    
+- (IBAction)didTapLogout:(id)sender {
+    [PFUser logOutInBackgroundWithBlock:^(NSError * _Nullable error) {
+        if (error != nil) {
+            NSLog(@"Error: %@", error.localizedDescription);
+        }
+        UIStoryboard *storyboard = [UIStoryboard storyboardWithName:@"Main" bundle:nil];
+        LoginViewController *loginViewController = [storyboard instantiateViewControllerWithIdentifier:@"LoginViewController"];
+        self.view.window.rootViewController = loginViewController;
+
+    }];
+}
+- (IBAction)didTapCameraIcon:(id)sender {
+    UIStoryboard *storyboard = [UIStoryboard storyboardWithName:@"Main" bundle:nil];
+    ImagePickerViewController *imagePickerViewController = [storyboard instantiateViewControllerWithIdentifier:composeSegue];
+    [self presentViewController:imagePickerViewController animated:YES completion:nil];
+    
+}
+
+-(NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
+    return self.posts.count;
+}
+
+-(UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
+    PostTableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"PostTableViewCell"];
+    Post *post = self.posts[indexPath.row];
+    cell.captionLabel.text = post.caption;
+    [cell setPost:post];
+    self.user = [PFUser currentUser];
+
+    PFFileObject *image = [post.author objectForKey:@"image"];
+    
+    cell.userLabel1.text = post.author.username;
+    cell.userLabel2.text = post.author.username;
+
+    [image getDataInBackgroundWithBlock:^(NSData *data, NSError *error) {
+        if (!data) {
+            return NSLog(@"%@", error);
+        }
+        cell.profileView.image = [UIImage imageWithData:data];
+    }];
+    cell.profileView.layer.cornerRadius = cell.profileView.frame.size.height/2;
+    
+    NSDateFormatter *formatter = [[NSDateFormatter alloc] init];
```

Comment:
I wouldn't have expected you to know this, but NSDateFormatter is expensive to create (https://stackoverflow.com/q/27321993), so it is common to create this once and save it as a static variable. Even if you didn't do that, I would suggest moving the formatter to line 174 where it's actually used, so that you don't create it unnecessarily for the cases handled on line 165 and 167.